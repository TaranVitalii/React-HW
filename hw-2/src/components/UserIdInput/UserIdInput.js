import React from "react";

export default props => {
  const onChange = e => {
    e.preventDefault();
    props.onChange(e.target.value);
  };
  return (
    <input
      type="text"
      name="UserId"
      placeholder="User id"
      onChange={onChange}
    />
  );
};
