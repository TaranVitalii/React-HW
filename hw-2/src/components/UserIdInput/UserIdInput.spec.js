import React from "react";
import UserIdInput from "./UserIdInput";
import { shallow } from "enzyme";

describe("<InputForm/>", () => {
  it("should render a UserId Input", () => {
    const wrapper = shallow(<UserIdInput />);

    expect(wrapper).toMatchSnapshot();
  });
  it("should pass a selected value to the onChange handler", () => {
    const value = 2;
    const onChange = jest.fn();
    const wrapper = shallow(<UserIdInput onChange={onChange} />);

    expect(wrapper).toMatchSnapshot();

    wrapper.find(".inputForm").simulate("change", {
      target: { value }
    });

    expect(onChange).toBeCalledWith(value);
  });
});
