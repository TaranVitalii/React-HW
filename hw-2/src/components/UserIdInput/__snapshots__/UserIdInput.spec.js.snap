// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm/> 1st test 1`] = `ShallowWrapper {}`;

exports[`<InputForm/> should pass a selected value to the onChange handler 1`] = `ShallowWrapper {}`;

exports[`<InputForm/> should render a UserId Input 1`] = `ShallowWrapper {}`;

exports[`should render a <InputForm/> 1st test 1`] = `ShallowWrapper {}`;
